sklearn/linear_model/_logistic.py

```diff
@@ -869,22 +872,40 @@ class LogisticRegression(LinearClassifierMixin, SparseCoefMixin, BaseEstimator):
         .. versionadded:: 0.19
            l1 penalty with SAGA solver (allowing 'multinomial' + L1)
 
-    dual : bool, default=False
-        Dual (constrained) or primal (regularized, see also
-        :ref:`this equation <regularized-logistic-loss>`) formulation. Dual formulation
-        is only implemented for l2 penalty with liblinear solver. Prefer dual=False when
-        n_samples > n_features.
-
-    tol : float, default=1e-4
-        Tolerance for stopping criteria.
+        .. deprecated:: 1.8
+           `penalty` was deprecated in version 1.8 and will be removed in 1.10.
+           Use `l1_ratio` instead. `l1_ratio=0` for `penalty='l2'`, `l1_ratio=1` for
+           `penalty='l1'` and `l1_ratio` set to any float between 0 and 1 for
+           `'penalty='elasticnet'`.
 
     C : float, default=1.0
         Inverse of regularization strength; must be a positive float.
         Like in support vector machines, smaller values specify stronger
-        regularization. For a visual example on the effect of tuning the `C` parameter
+        regularization. `C=np.inf` results in unpenalized logistic regression.
+        For a visual example on the effect of tuning the `C` parameter
         with an L1 penalty, see:
         :ref:`sphx_glr_auto_examples_linear_model_plot_logistic_path.py`.
 
+    l1_ratio : float, default=0.0
+        The Elastic-Net mixing parameter, with ``0 <= l1_ratio <= 1``. Only
+        used if ``penalty='elasticnet'``. Setting ``l1_ratio=0`` is equivalent
```

Comment:
I got a bit surprised with the "only used if penalty='elasticnet'".

Could we move it at the end of the paragraph instead and mention that `l1_ratio` has an effect with `penalty` is let as its default or that one force `penalty="elasticnet"` (but it is deprecated).